### PR TITLE
Adding Suspend/Resume notification code()

### PR DESCRIPTION
**Pull request by huzaifa-d**

### CODE REVIEW

File: dev/PowerNotifications/PowerNotifications.h [hunk 1]
```diff
@@ -592,6 +605,41 @@ namespace winrt::Microsoft::ProjectReunion::factory_implementation
             RaiseEvent(systemAwayModeStatusFunc);
         }
 
+        //SystemSuspend Functions
+        winrt::Microsoft::ProjectReunion::SystemSuspendStatus SystemSuspendStatus()
+        {
+            return static_cast<winrt::Microsoft::ProjectReunion::SystemSuspendStatus>(m_systemSuspendStatus);
```
Comment (brialmsft) [inline dev/PowerNotifications/PowerNotifications.h:611]: What happens if somebody calls this before anyone has registered a listener and had it fire? Isn't this undefined?
Is the expectation that this is only legal to call from inside of a callback? If so, that's different then other propgets on the same class, which seems potentially confusing.

Comment (huzaifa-d) [reply dev/PowerNotifications/PowerNotifications.h:616]: Added uninitialized enum value to handle that. Will be finalized by Andrew/Jon as discussed.

File: dev/PowerNotifications/PowerNotifications.h [hunk 2]
```diff
@@ -608,6 +608,11 @@ namespace winrt::Microsoft::ProjectReunion::factory_implementation
         //SystemSuspend Functions
         winrt::Microsoft::ProjectReunion::SystemSuspendStatus SystemSuspendStatus()
         {
+Addi            // Review: Agree with this usage-only-after-subscription enforcement?
```
Comment (brialmsft) [inline dev/PowerNotifications/PowerNotifications.h:611]: Unintended text "Addi"?

File: dev/PowerNotifications/PowerNotifications.h [hunk 3]
```diff
@@ -608,6 +608,11 @@ namespace winrt::Microsoft::ProjectReunion::factory_implementation
...
+            if (m_systemSuspendStatus == SystemSuspendStatus::Uninitialized)
+            {
+                throw winrt::hresult_error(E_FAIL, L"There are no active subscriptions on SystemSuspendStatus");
```
Comment (brialmsft) [inline dev/PowerNotifications/PowerNotifications.h:614]: I think "Only callable during SystemSuspendStatusChanged callback" is more accurate.

File: dev/PowerNotifications/PowerNotifications.h (hunk 2 above)
Comment (huzaifa-d) [reply dev/PowerNotifications/PowerNotifications.h]: Yes. Fixed.

File: dev/PowerNotifications/PowerNotifications.h (hunk 3 above)
Comment (brialmsft) [reply dev/PowerNotifications/PowerNotifications.h:614]: I was referring to the error message on line 614. Even if you have an active subscription, you're blocked from calling this method from outside of a callback, right? That's why I'm recommending that the message be clarified.

**Review by brialmsft — APPROVED** [review]

**Pull request merged**